### PR TITLE
Changed the way values are updated

### LINKED ISSUE #713: Remove lazy update of oracle price
Should just update the aggregated price upon new feed.

### PR DESCRIPTION
**Pull request by syan095**
Oracle prices are updated on price feed, instead of "get".

Closes #713

### CODE REVIEW

File: oracle/src/lib.rs [hunk 1]
```diff
@@ -247,7 +230,12 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				timestamp: now,
 			};
 			RawValues::<T, I>::insert(&who, &key, timestamped);
-			IsUpdated::<T, I>::remove(&key);
+
+			let combined = Self::combined(key);
+			if combined.is_some() {
```
Comment (xlc) [inline oracle/src/lib.rs:235]: `if let Some(combined) = combined` and no need unwrap

File: oracle/src/lib.rs [hunk 2]
```diff
@@ -187,26 +181,15 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	///
 	/// Note this will update values storage if has update.
```
Comment (shaunxw) [inline oracle/src/lib.rs:182]: docstring needs update

File: oracle/src/lib.rs [hunk 3]
```diff
@@ -187,26 +181,15 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
 	pub fn get(key: &T::OracleKey) -> Option<TimestampedValueOf<T, I>> {
-		if Self::is_updated(key) {
-			<Values<T, I>>::get(key)
-		} else {
-			let timestamped = Self::combined(key)?;
-			<Values<T, I>>::insert(key, timestamped.clone());
-			IsUpdated::<T, I>::insert(key, true);
-			Some(timestamped)
-		}
+		<Values<T, I>>::get(key)
```
Comment (shaunxw) [inline oracle/src/lib.rs:184]: Self::values(key)

File: oracle/src/lib.rs [hunk 4]
```diff
@@ -187,26 +181,15 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
 	}
 
 	/// Returns fresh combined value if has update, or latest combined
 	/// value.
 	///
 	/// This is a no-op function which would not change storage.
 	pub fn get_no_op(key: &T::OracleKey) -> Option<TimestampedValueOf<T, I>> {
```
Comment (shaunxw) [inline oracle/src/lib.rs:191]: We could probably remove `get_no_op` now, as `get` is currently no-op.

**Comment by xlc** [comment]
`get_all_values` can also be simplified

**Comment by syan095** [comment]
> `get_all_values` can also be simplified

Done

**Review by xlc — APPROVED** [review]

**Pull request merged**